src/augmentation.py

```diff
@@ -166,4 +166,166 @@ def _perspective_transform_augment_images(self, images, random_state, parents, h
+        result = []
+        for i, image in enumerate(images):
+            image_pad = self._pad(image)
+            result.append(image_pad)
+        return result
+
+    def _augment_keypoints(self, keypoints_on_images, random_state, parents, hooks):
+        result = []
+        return result
+
+    def _pad(self, img):
+        img_ = img.copy()
+
+        if self._is_expanded_grey_format(img):
+            img_ = np.squeeze(img_, axis=-1)
+
+        h_pad, w_pad = self.pad
+        img_ = cv2.copyMakeBorder(img_.copy(), h_pad, h_pad, w_pad, w_pad, PadFixed.PAD_FUNCTION[self.pad_method])
+
+        if self._is_expanded_grey_format(img):
+            img_ = np.expand_dims(img_, axis=-1)
+
+        return img_
+
+    def get_parameters(self):
+        return []
+
+    def _is_expanded_grey_format(self, img):
+        if len(img.shape) == 3 and img.shape[2] == 1:
+            return True
+        else:
+            return False
+
+
+class RandomCropFixedSize(iaa.Augmenter):
+    def __init__(self, px=None, name=None, deterministic=False, random_state=None):
+        super(RandomCropFixedSize, self).__init__(name=name, deterministic=deterministic, random_state=random_state)
+        self.px = px
+        if isinstance(self.px, tuple):
+            self.px_h, self.px_w = self.px
+        elif isinstance(self.px, int):
+            self.px_h = self.px
+            self.px_w = self.px
+        else:
+            raise NotImplementedError
+
+    def _augment_images(self, images, random_state, parents, hooks):
+
+        result = []
+        seeds = random_state.randint(0, 10 ** 6, (len(images),))
+        for i, image in enumerate(images):
+            seed = seeds[i]
+            image_cr = self._random_crop(seed, image)
+            result.append(image_cr)
+        return result
+
+    def _augment_keypoints(self, keypoints_on_images, random_state, parents, hooks):
+        result = []
+        return result
+
+    def _random_crop(self, seed, image):
+        height, width = image.shape[:2]
+
+        np.random.seed(seed)
+        if height > self.px_h:
+            crop_top = np.random.randint(height - self.px_h)
+        elif height == self.px_h:
+            crop_top = 0
+        else:
+            raise ValueError("To big crop height")
+        crop_bottom = crop_top + self.px_h
+
+        np.random.seed(seed + 1)
+        if width > self.px_w:
+            crop_left = np.random.randint(width - self.px_w)
+        elif width == self.px_w:
+            crop_left = 0
+        else:
+            raise ValueError("To big crop width")
+        crop_right = crop_left + self.px_w
+
+        if len(image.shape) == 2:
+            image_cropped = image[crop_top:crop_bottom, crop_left:crop_right]
+        else:
+            image_cropped = image[crop_top:crop_bottom, crop_left:crop_right, :]
+        return image_cropped
+
+    def get_parameters(self):
+        return []
+
+
+class PadToFitInNetwork(iaa.Augmenter):
```

Comment:
@apyskir not sure about this name. How about we call it InferencePad or DivisibleSizePad?

Reply:
@jakubczakon OK let's keep `InferencePad`